frontend/class-opengraph-image.php

```diff
@@ -196,11 +196,22 @@ public function add_image( $attachment ) {
 	 * If the frontpage image exists, call add_image.
 	 */
 	private function set_front_page_image() {
+
+		$show_on_front = get_option( 'show_on_front' );
+
+		// If frontpage is a simple page, call the set_singular_image function.
+		if ( $show_on_front !== 'posts' ) {
+			$this->set_singular_image();
+
+			return;
```

Comment:
I think we also should have a fallback to the `og_frontpage_image` when no image has been found. Please wrap this return with:

```php
if ( $this->has_images() { 
   return;
}
```